library/Hasql/Private/Prelude.hs

```diff
@@ -51,7 +53,12 @@ import Contravariant.Extras as Exports
 -- either
 -------------------------
 import Control.Monad.Trans.Either as Exports
+-- prefer implementations in base (via base-prelude) over the ones in either
+#if MIN_VERSION_base(4,10,0)
+import Data.Either.Combinators as Exports hiding (fromLeft, fromRight)
```

Comment:
base < 4.10 does not define `fromLeft` / `fromRight`, so they'd be missing inside `hasql`.
A quick `grep` however reveals that neither of them is used inside the whole package.

Under these conditions I concur: it's better to leave out CPP. Will update the PR.